### PR TITLE
docs: add Kakao Auth docs

### PR DESCRIPTION
**Pull request by hwr12**
## What kind of change does this PR introduce?

Added Kakao Auth docs
The kakao auth functionality has been added by this PR
https://github.com/supabase/gotrue/pull/834#issuecomment-1543499472

## Additional context

Google Chrome-2023 05 11 18 56-001111
Google Chrome-2023 05 11 18 57-001113 
Google Chrome-2023 05 11 18 57-001115

### CODE REVIEW

**Comment by hwr12** [comment]
Hey @kangmingtay !
I've added docs for Kakao Auth.
Could you review it?

File: apps/docs/pages/guides/auth/social-login/auth-kakao.mdx
Comment (kangmingtay) [inline apps/docs/pages/guides/auth/social-login/auth-kakao.mdx:36]:
```suggestion
- Click `Save` at the bottom right.
```

File: apps/docs/pages/guides/auth/social-login/auth-kakao.mdx
Comment (kangmingtay) [inline apps/docs/pages/guides/auth/social-login/auth-kakao.mdx:67]:
```suggestion
async function signInWithKakao() {
```

File: apps/docs/pages/guides/auth/social-login/auth-kakao.mdx
Comment (kangmingtay) [inline apps/docs/pages/guides/auth/social-login/auth-kakao.mdx:47]: can you also indicate where to add the callback URL on kakao please?

Comment (kangmingtay) [reply apps/docs/pages/guides/auth/social-login/auth-kakao.mdx:47]: We would just need a sentence to tell the user to add the callback url to this section

image

**Review by kangmingtay — CHANGES_REQUESTED** [review]
Hey @hwr12, thanks for getting this up so quickly! I've left a couple of comments but overall, this is really good. Once we merge the Kakao OAuth PR (thanks @esinx!) into gotrue, we'll merge this one in.

**Comment by kangmingtay** [comment]
@hwr12 can you also add another section for the following steps:

* Ensure that Kakao Login is enabled in the application
* Ensure that the following scopes are set under the "Consent Items": `account_email`, `profile_image`, `profile_nickname` 
image

File: apps/docs/pages/guides/auth/social-login/auth-kakao.mdx
Comment (hwr12) [reply apps/docs/pages/guides/auth/social-login/auth-kakao.mdx:47]: Yup. Will it be good to put the sentence at the bottom inside `Find your callback URL` section?

**Comment by hwr12** [comment]
> @hwr12 can you also add another section for the following steps:
> 
> * Ensure that Kakao Login is enabled in the application
> * Ensure that the following scopes are set under the "Consent Items": `account_email`, `profile_image`, `profile_nickname`
> 
> image

 Account_email, profile_image, profile_nickname
 These are needed because of the gotrue codes that uses these values?

**Comment by hwr12** [comment]
@kangmingtay What are the Vercel workflows for? Do I need to pass it?

**Comment by kangmingtay** [comment]
@hwr12 nah don't worry about those, but can you please format the code with prettier?

**Comment by hwr12** [comment]
@kangmingtay I am kind of new to contributing open source. Sorry for the unformatted code. Thanks for helping me, I really appreciate it!

**Review by kangmingtay — APPROVED** [review]
LGTM, we should be able to merge this in once we've deployed the latest version of supabase-auth to all projects (so that it includes the kakao update)

**Comment by artifacts3** [comment]
please, i really can`t wait it

**Comment by gregnr** [comment]
@kangmingtay safe to merge at this point?

**Comment by kangmingtay** [comment]
@gregnr nope not yet, i'm waiting on the deployment to roll to all projects before we merge this in

**Comment by kokihoon** [comment]
When will I become a merge?

**Comment by kangmingtay** [comment]
@kokihoon once we've deployed minimally this version of gotrue on the platform: https://github.com/supabase/gotrue/releases/tag/v2.67.0

we're aiming to deploy the change to the platform at the end of the week! 

the current version on the platform is [v2.66.0](https://github.com/supabase/gotrue/releases/tag/v2.66.0)

**Comment by hwr12** [comment]
> @kokihoon once we've deployed minimally this version of gotrue on the platform: https://github.com/supabase/gotrue/releases/tag/v2.67.0
> 
> we're aiming to deploy the change to the platform at the end of the week!
> 
> the current version on the platform is [v2.66.0](https://github.com/supabase/gotrue/releases/tag/v2.66.0)

This is a great news as many developers have been waiting for this feature!! 🎉

File: apps/docs/pages/guides/auth/social-login/auth-kakao.mdx
Comment (kangmingtay) [inline apps/docs/pages/guides/auth/social-login/auth-kakao.mdx:17]:
```suggestion
- Generating the `Client secret code` - this will serve as the `client_secret`.
```

**Review by gregnr — APPROVED** [review]

**Pull request merged**
